### PR TITLE
Make reading VERSION file optional

### PR DESCRIPTION
**Pull request by j8r**
Fix https://github.com/crystal-lang/crystal/issues/7602
Fix https://github.com/crystal-lang/distribution-scripts/issues/35

### CODE REVIEW

File: src/compiler/crystal/config.cr [hunk 1]
```diff
@@ -5,7 +5,7 @@ module Crystal
     end
 
     def self.version
-      {{ read_file("#{__DIR__}/../../../VERSION").chomp }}
+      {{ (read_file?("#{__DIR__}/../../../VERSION") || "").chomp }}
```
Comment (Sija) [inline src/compiler/crystal/config.cr]: That's incorrect. If there's no `VERSION` file it should return `nil`.

Comment (j8r) [reply src/compiler/crystal/config.cr]: That's how CRYSTAL_PATH has done

**Comment by bcardiff** [comment]
I would prefer to include the VERSION file in the distribution packages.

**Comment by j8r** [comment]
No @bcardiff , that's the purpose of the file. If we download the crystal source as a tar.gz, we have the version inside.

**Comment by asterite** [comment]
What's wrong with including the `VERSION` file in our distribution?

**Comment by j8r** [comment]
Because not everybody use your distribution scripts, packages like Alpine, Arch Linux and others. Everybody must use the source code from crystal-lang/crystal, some use directly https://github.com/crystal-lang/crystal/archive/0.28.0.tar.gz.

**Comment by ysbaddaden** [comment]
This isn't a good example because the release tarballs do have the `VERSION` file, since they're snapshots of the repository. Do you have examples where the `VERSION` file is _actually_ missing?

**Comment by j8r** [comment]
I was responding to @asterite and @bcardiff @ysbaddaden, of course the `VERSION` file is present. If the file is moved to distribution-scripts, it will no longer be present when packaging for Alpine and Arch Linux, then no version will be available.

For your question about when `VERSION` is missing, look at the two issues in the description (you can ask to @bew about its usecase)

**Comment by bcardiff** [comment]
The `VERSION` file is part of the source. There is no need to make it an optional dependency.
If the file would have been in the ~/src directory it would have worked out of the box.
distribution packages already might already be packaging docs, samples, license, stdlib, etc.

They need to be updated to include the VERSION file to support the scenario of requiring the compiler.

**Comment by j8r** [comment]
I misunderstood. I was thinking that including `VERSION` meant moving it from the repo to distribution scripts.

**Comment by RX14** [comment]
Arch installs the `src` directory to `/usr/lib/crystal`. We shouldn't require any files *outside* the `src` directory to compile what's inside `src` because we cannot assume that they're there.

A simple fix is moving `VERSION` to `src/VERSION`.

**Pull request closed without merging**